src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,147 @@
+        showToUser(HELLO_MESSAGE);
+        showToUser(SEPARATOR);
+    }
+
+    private static String getUserInput() {
+        return SCANNER.nextLine();
+    }
+
+    private static void executeCommand(String userInputString) throws DukeException {
+        showToUser(SEPARATOR);
+        final String[] commandTypeAndParams = splitCommandWordAndArgs(userInputString);
+        final String commandType = commandTypeAndParams[0];
+        final String commandArgs = commandTypeAndParams[1];
+        switch (commandType) {
+        case COMMAND_LIST_WORD:
+            listTask();
+            break;
+        case COMMAND_MARK_WORD:
+            markAsDone(commandArgs);
+            break;
+        case COMMAND_UNMARK_WORD:
+            markAsNotDone(commandArgs);
+            break;
+        case COMMAND_EXIT_WORD:
+            exitProgram();
+            break;
+        case COMMAND_ADD_TODO_WORD:
+        case COMMAND_ADD_DEADLINE_WORD:
+        case COMMAND_ADD_EVENT_WORD:
+            addTask(commandType, commandArgs);
+            break;
+        default:
+            throw new UserInputException(INVALID_TASK_TYPE);
+        }
+        showToUser(SEPARATOR);
+    }
+
+    public static void showToUser(String... message) {
+        for (String m : message) {
+            System.out.println(LINE_PREFIX + m);
+        }
+    }
+
+    private static void addTask(String newTaskType, String newTaskInfo) throws DukeException {
+        switch (newTaskType) {
+        case COMMAND_ADD_TODO_WORD:
+            TASK_LIST[taskCount] = new Todo(newTaskInfo);
+            break;
+        case COMMAND_ADD_DEADLINE_WORD:
+
+            TASK_LIST[taskCount] = new Deadline(newTaskInfo);
+            break;
+        case COMMAND_ADD_EVENT_WORD:
+
+            TASK_LIST[taskCount] = new Event(newTaskInfo);
+            break;
+        default:
+            throw new UserInputException(INVALID_TASK_TYPE);
+        }
+        showToUser("Got it. I've added this task:",
+                TASK_LIST[taskCount].toString(),
+                "Now you have " + (taskCount + 1) + (taskCount + 1 == 0 ? " task" : " tasks") + " in the list.");
+        taskCount += 1;
+    }
+
+    private static void markAsDone(String taskNumber) {
+        Task taskToBeMarked = TASK_LIST[Integer.parseInt(taskNumber) - 1];
+        taskToBeMarked.toggleDone();
+        showToUser("Nice! I've marked this task as done:");
+        showToUser(taskToBeMarked.toString());
+    }
+
+    private static void markAsNotDone(String taskNumber) {
+        Task taskToBeUnmarked = TASK_LIST[Integer.parseInt(taskNumber) - 1];
+        taskToBeUnmarked.toggleDone();
+        showToUser("OK, I've marked this task as not done yet:");
+        showToUser(taskToBeUnmarked.toString());
+    }
+
+    private static void listTask() {
+        showToUser("Here are the tasks in your list:");
+        for (int i = 0; i < taskCount; i++) {
+            showToUser((i + 1) + ". " + TASK_LIST[i].toString());
```

Comment:
Very good application of polymorphism. You can also apply the same concept when you work on the future levels!